include/aws/http/request_response.h

```diff
@@ -194,6 +194,42 @@ typedef void(aws_http_on_stream_complete_fn)(struct aws_http_stream *stream, int
  */
 typedef void(aws_http_on_stream_destroy_fn)(void *user_data);
 
+/**
+ * Tracing metrics for aws_http_stream.
+ * Data maybe not be available if the data of stream was never sent/received before it completes.
+ */
+struct aws_http_stream_metrics {
```

Comment:
a couple more things to consider adding: 
- tls overhead
- connection acquisition (do we have connection in the pool available or we have to get a new one)
- retries if we have visibility into http level retries
- number of bytes sent/received
- errors/status codes

Reply:
- tls overhead is more a connection related metrics, we can add it as a separate connection level thing. 
- connection acquisition (do we have connection in the pool available or we have to get a new one)
  - that's connection manager level, we have `struct aws_http_manager_metrics` [here](https://github.com/awslabs/aws-c-http/blob/main/include/aws/http/connection_manager.h#L31), not really telling people the connection is acquired from poll or created, but provide an overall status of the connection 
- I don't think we have retries from http level (?)
-  number of bytes sent/received
  - I think that's something we can consider, even though we have callbacks for headers/body, the un-decoded bytes length can still be helpful.
- errors/status codes
  - There are complete callback that we provide the error code.

Reply:
What about how much time we spend awaiting input?

Reply:
don't we already have tls level metrics?